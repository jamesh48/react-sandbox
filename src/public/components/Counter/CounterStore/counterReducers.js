// import {  } from './Actions.js';

const combineReducers = (slices) => (state, action) => Object.keys(slices).reduce((acc, prop) => ({
  ...acc,
  [prop]: slices[prop](acc[prop], action),
}),
  state
);

const reduceReducers = (...reducers) => {
  return (state, action) => reducers.reduce((acc, nextReducer) => nextReducer(acc, action), state)
};

const count = (state = 0, { type, payload }) => {
  switch (type) {
    case 'INCREMENT COUNTER':
      return state + 1;
    case 'DECREMENT COUNTER':
      return state - 1;
    default:
      return state;
  };
};

export default combineReducers({ count });