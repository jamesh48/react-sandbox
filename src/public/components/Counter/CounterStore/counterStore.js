import React from 'react';
import CombinedReducers from './counterReducers.js';
const CounterStoreContext = React.createContext();
const initialState = { count: 0 };
const CounterStoreProvider = ({ children }) => {
  const [state, dispatch] = React.useReducer(CombinedReducers,
    initialState);
  const store = React.useMemo(() => [state, dispatch], [state]);
  return (
    <CounterStoreContext.Provider value={store}>{children}</CounterStoreContext.Provider>
  );
};
const useCounterContext = () => {
  const context = React.useContext(CounterStoreContext);
  if (context === undefined) {
    throw new Error('useCounterContext must be used within a CounterProvider')
  }
  return context;
};
export { useCounterContext, CounterStoreProvider };