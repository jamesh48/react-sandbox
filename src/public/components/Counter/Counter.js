import React from 'react'
import 'GlobalSCSS';

import { useCounterContext } from 'CounterStore';

export default () => {
  const [{ count }, counterDispatch] = useCounterContext();

  return (
    <div className='component-container' id='counter-root'>
      <div>Count: {count}</div>
      <span style={{ display: 'flex' }}>
        <button onClick={() => counterDispatch({ type: 'INCREMENT COUNTER' })}>+</button>
        <button onClick={() => counterDispatch({ type: 'DECREMENT COUNTER' })}>-</button>
      </span>
    </div>
  );
};