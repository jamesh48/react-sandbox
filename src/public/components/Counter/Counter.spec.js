import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import Counter from './Counter.js';
import { CounterStoreProvider } from 'CounterStore';

describe('Counter Component', () => {

  it('Renders the Counter Component', () => {
    const wrapper = mount(
      <CounterStoreProvider>
        <Counter />
      </CounterStoreProvider>
    );
    expect(wrapper.find('#counter-root').length).toEqual(1);
  });

  test('counter snapshot renders', () => {
    const component = renderer.create();
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

});