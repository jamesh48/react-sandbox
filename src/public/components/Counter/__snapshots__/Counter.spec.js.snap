// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter Component counter snapshot renders 1`] = `null`;
